src/components/zones/ZonesView.tsx

```diff
@@ -117,48 +179,86 @@ export default function ZonesView({
               </Flex>
             }
           />
-        </Routes>
-      </Flex>
+        </Routes> */}
+
       {zones.map((z) => {
         return (
-          <Rnd
+          <ZoneFrame
             key={z.id}
-            default={{
-              x: z.x,
-              y: z.y,
-              width: z.width,
-              height: z.height,
-            }}
-            className={cx(zoneStyle, {
-              [selectedZoneStyle]: z.status === "EDITING",
-            })}
-            onMouseDown={() => dispatch(zoneActiveToggled(z.id))}
-            enableResizing={z.status === "EDITING" && !disableEdition}
-            disableDragging={disableEdition}
-            onResizeStop={(e, direction, ref, delta, position) => {
-              const newZone = {
-                id: z.id,
-                x: position.x,
-                y: position.y,
-                width: z.width + delta.width,
-                height: z.height + delta.height,
-              };
-              dispatch(zoneUpdated(newZone));
-            }}
-            onDragStop={(e, d) => {
-              const newPos = {
-                id: z.id,
-                x: d.x,
-                y: d.y,
-              };
-              dispatch(zoneUpdated(newPos));
-            }}
-            resizeHandleComponent={handleComp}
-          >
-            <p className={aboveZoneStyle}>{z.name}</p>
-          </Rnd>
+            zone={z}
+            disableEdition={disableEdition}
+            zoom={zoom}
+          />
         );
       })}
     </Flex>
   );
 }
+
+interface ZoneFrameProps {
+  //RCmenustate:RightClickMenuState;
+  zone: Zone;
+  disableEdition: boolean;
+  zoom: number;
+}
+function ZoneFrame({
+  //RCmenustate,
+  zone,
+  disableEdition,
+  zoom,
+}: ZoneFrameProps) {
+  const dispatch = useDispatch();
+
+  return (
+    <Rnd
+      key={zone.id}
+      id={zone.id}
+      default={{
+        x: zone.x,
+        y: zone.y,
+        width: zone.width,
+        height: zone.height,
+      }}
+      size={{
+        width: zone.width * zoom,
+        height: zone.height * zoom,
+      }}
+      position={{
+        x: zone.x * zoom,
+        y: zone.y * zoom,
+      }}
+      className={
+        "rightClickable " +
+        cx(zoneStyle, {
+          [selectedZoneStyle]: zone.status === "EDITING",
+        })
+      }
+      onMouseDown={() => dispatch(zoneActiveToggled(zone.id))}
```

Comment:
Right-clicking on the active zone unselect it and does not show the right-click menu. Maybe testing for the clicked button before toggling the zone should do the trick ?
`onMouseDown={(e) => {
        const MOUSE_MAIN_BUTTON = 0;
        const MOUSE_RIGHT_BUTTON = 2;
        if (
          e.button === MOUSE_MAIN_BUTTON ||
          (e.button === MOUSE_RIGHT_BUTTON && zone.id !== selectedZone?.id)
        ) {
          dispatch(zoneActiveToggled(zone.id));
        }
      }}`